From d4c049059f8df6890b945c96f2ac767b0ed6826f Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 22 Oct 2013 16:45:00 +0000
Subject: GtkScrolledWindow: Disconnect from frame clock properly

The tick callback IDs from GtkWidget aren't timeouts, so
use the correct function to disconnect from them.

Spotted by Benjamin Otte <otte@redhat.com>

https://bugzilla.gnome.org/show_bug.cgi?id=710666
---
diff --git a/gtk/gtkscrolledwindow.c b/gtk/gtkscrolledwindow.c
index 99dab9e..9743be7 100644
--- a/gtk/gtkscrolledwindow.c
+++ b/gtk/gtkscrolledwindow.c
@@ -1145,7 +1145,7 @@ gtk_scrolled_window_set_kinetic_scrolling (GtkScrolledWindow *scrolled_window,
         }
       if (priv->deceleration_id)
         {
-          g_source_remove (priv->deceleration_id);
+          gtk_widget_remove_tick_callback (GTK_WIDGET (scrolled_window), priv->deceleration_id);
           priv->deceleration_id = 0;
         }
     }
@@ -1252,7 +1252,7 @@ gtk_scrolled_window_destroy (GtkWidget *widget)
     }
   if (priv->deceleration_id)
     {
-      g_source_remove (priv->deceleration_id);
+      gtk_widget_remove_tick_callback (widget, priv->deceleration_id);
       priv->deceleration_id = 0;
     }
 
--
cgit v0.9.2

